contracts/contracts/Tally.sol

```diff
@@ -244,9 +254,48 @@ contract Tally is Ownable, SnarkCommon, CommonUtilities, Hasher {
     tally[1] = hashLeftRight(_totalSpent, _totalSpentSalt);
     tally[2] = _perVOSpentVoiceCreditsHash;
 
+    isValid = isQv
+      ? verifyQvSpentVoiceCredits(_totalSpent, _totalSpentSalt, _resultCommitment, _perVOSpentVoiceCreditsHash)
+      : verifyNonQvSpentVoiceCredits(_totalSpent, _totalSpentSalt, _resultCommitment);
+  }
+
+  /// @notice Verify the number of spent voice credits for QV from the tally.json
+  /// @param _totalSpent spent field retrieved in the totalSpentVoiceCredits object
+  /// @param _totalSpentSalt the corresponding salt in the totalSpentVoiceCredit object
+  /// @param _resultCommitment hashLeftRight(merkle root of the results.tally, results.salt) in tally.json file
+  /// @param _perVOSpentVoiceCreditsHash hashLeftRight(merkle root of the no spent voice credits per vote option, salt)
+  /// @return isValid Whether the provided values are valid
+  function verifyQvSpentVoiceCredits(
+    uint256 _totalSpent,
+    uint256 _totalSpentSalt,
+    uint256 _resultCommitment,
+    uint256 _perVOSpentVoiceCreditsHash
+  ) private view returns (bool isValid) {
```

Comment:
can we make it internal in case a contract inheriting from Tally might need direct access?